### ISSUE TITLE
Kodi 18.7

### ISSUE THREAD

**Issue by HinataNekoNya** [garbear/kodi-steamlink#37]
Kodi 18.7 has arrived, can you just update it? 

Thanks for awsome

HinataNekoNya

**Comment by n00bhax** [comment]
Will this still be updated or is there a fork somewhere with a more recent version?

**Comment by n00bhax** [comment]
It seems like mittler worked on it, but his branch doesn't have a release and I don't know how to build it.

**Comment by garbear** [comment]
Correct, I got Kodi 18 running on the Steam Link, but the performance was bleak, so I abandoned the port. The build instructions are documented, so I can help you through the build process if you'd like.

**Comment by n00bhax** [comment]
Would it be worth it? Since you mentioned poor performance. If so, some help with building would be appreciated :). I'm trying to use jellyfin on my link btw.

**Comment by garbear** [comment]
Unfortunately I don't think so, due to the SL's real-time streaming API, there are some sync issues between audio and video, so poor performance beside it's not the best experience.

**Comment by n00bhax** [comment]
So even the v17.6 sucks? Guess I'll just keep using my shitty Chromecast then😢

**Comment by leahcim3** [comment]
> So even the v17.6 sucks? Guess I'll just keep using my shitty Chromecast thencry

The chromecast V1 has more or less the same hardware - well the Marvell SoC used.

Google probably use hardware decoding though so it'll presumably work better.

If you grep the source here you'll see a bunch of steamlink folders Valve added and, so far as I can tell, they just cut and paste some code from ffmpeg to do h264 decoding in software. Hence why it's slow and choppy.

To get this to work it would need the steamlink code to interface with kodi to use whatever libraries / apis (proprietary or otherwise) that are on the steamlink (probably in the /lib folder) that do H264 hardware decoding.

That must exist because the streamingclient app (which runs when you're streaming a game on the steamlink) uses about 10% cpu for 1080p/60 fps video. Whereas this uses about 70-80% for 720p video that's choppy.

Whether valve couldn't be bothered or weren't allowed to implement this properly is unknown, but to save anyone the bother - this version of kodi is a complete waste of space as implemented. Not sure why valve uploaded it TBH.

Sucks really because, as I said, it's more or less a v1 Chromecast - so if that works, well this could have too.
